lib/phoenix_live_view/async.ex

```diff
@@ -50,20 +53,36 @@ defmodule Phoenix.LiveView.Async do
     :ok
   end
 
+  defp validate_function_env(func, op) do
+    {:env, variables} = Function.info(func, :env)
```

Comment:
Maybe it is still worth gating this behind a compile_env flag?

Reply:
Maybe. I have no idea what the consequences could be, but the docs are quite clear.

> Note: this function must be used only for debugging purposes.

If we're putting it behind an application env flag, what name would you propose? It's not really checking for expensive assigns, so I wouldn't necessarily use the same as #3157. But maybe both could use the same flag, so something like

`:enable_runtime_code_checks`

that could allow even more checks in the future and then be enabled for new apps in dev env by default.

Reply:
:enable_expensive_runtime_checks ?